pkg/mcs/discovery/discover.go

```diff
@@ -35,3 +45,77 @@ func Discover(cli *clientv3.Client, clusterID, serviceName string) ([]string, er
 	}
 	return values, nil
 }
+
+// GetMCSPrimary returns the primary member of the specified service name.
+func GetMCSPrimary(name string, client *clientv3.Client, keyspaceGroupID string) (*pdpb.Member, int64, error) {
+	keyspaceGroupIDKey := utils.DefaultKeyspaceGroupID
+	if keyspaceGroupID != "" {
+		isValid := func(id uint32) bool {
+			return id >= utils.DefaultKeyspaceGroupID && id <= utils.MaxKeyspaceGroupCountInUse
+		}
+
+		keyspaceGroupID, err := strconv.ParseUint(keyspaceGroupID, 10, 64)
+		if err != nil || !isValid(uint32(keyspaceGroupID)) {
+			return nil, 0, errors.Errorf("invalid keyspace group id %d", keyspaceGroupID)
+		}
+		keyspaceGroupIDKey = uint32(keyspaceGroupID)
+	}
+
+	primaryPath, err := getMCSPrimaryPath(name, client, keyspaceGroupIDKey)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	return election.GetLeader(client, primaryPath)
+}
+
+// GetMembers returns all the members of the specified service name.
+func GetMembers(name string, client *clientv3.Client) (*clientv3.TxnResponse, error) {
```

Comment:
Can we handle the corresponding data structure internally in this function instead of directly returning an etcd TxnResponse?

Reply:
Good catch, some stale implementation 0 0